File: Books/Array/JS/Sorts/bubbleSort.js
class BubbleSort {
    a;
    nElems;

    constructor(size) {
        this.a = new Array(size);
        this.nElems = 0;
    }

    insert(value) {
        this.a[this.nElems] = value;
        this.nElems++;
    }

    display() {
        const list = [];
        for (let i = 0; i < this.nElems; i++) {
            list.push(this.a[i] + "")
        }
        console.log(list.join(' '));
    }

    sort() {
        let out;
        let inner;

        for (out = this.nElems - 1; out > 0; out--) {
            for (inner = 0; inner < out; inner++) {
                if (this.a[inner] > this.a[inner + 1]) {
                    const temp = this.a[inner];
                    this.a[inner] = this.a[inner + 1]
                    this.a[inner + 1] = temp;
                }
            }
        }
    }
}

function BibbleSortApp() {
    const maxSize = 100;
    const arr = new BubbleSort(maxSize)

    arr.insert(44);
    arr.insert(88);
    arr.insert(33);
    arr.insert(66);
    arr.insert(99);
    arr.insert(55);
    arr.insert(22);
    arr.insert(77);
    arr.insert(11);
    arr.display();

    arr.bubbleSort();
    arr.display();

}

BibbleSortApp();

File: Books/Sorts/Java/bubbleSort.java
package Books.Sorts.Java;

class BubbleSortApp {
    public static void main(String[] args) {
        int maxSize = 100;
        BubbleSort arr;

        arr = new BubbleSort(maxSize);

        arr.insert(44);
        arr.insert(88);
        arr.insert(33);
        arr.insert(66);
        arr.insert(99);
        arr.insert(55);
        arr.insert(22);
        arr.insert(77);
        arr.insert(11);
        arr.display();

        arr.sort();
        arr.display();
    }
}

class BubbleSort {
    private long[] a;
    private int nElems;

    public BubbleSort(int max) {
        a = new long[max];
        nElems = 0;
    }

    public void insert(long value) {
        a[nElems] = value;
        nElems++;
    }

    public void display() {
        for (int j = 0; j < nElems; j++)
            System.out.print(a[j] + " ");
        System.out.println("");
    }

    public void sort() {
        int out, in;
        for (out = nElems - 1; out > 0; out--) {
            for (in = 0; in < out; in++) {
                if (a[in] > a[in + 1]) {
                    long temp = a[in];
                    a[in] = a[in + 1];
                    a[in + 1] = temp;
                }
            }
        }
    }
};
